frontend/src/pages/RegisterInvestor.test.tsx

```diff
@@ -0,0 +1,82 @@
+import React from "react";
+import { render, screen, fireEvent } from "@testing-library/react";
+import RegisterInvestor from "./RegisterInvestor";
+
+const mockFetch = jest.fn((url, options) => {
+  const body = options && typeof options === "object" ? JSON.parse((options as any).body) : {};
+  if (body.email === "bob@investor.com") {
+    return Promise.resolve({
+      status: 400,
+      json: () => Promise.resolve({ email: ["Ця електронна пошта вже зареєстрована"] }),
+    });
+  }
+  return Promise.resolve({
+    status: 201,
+    json: () => Promise.resolve({ message: "success" }),
+  });
+});
+
+beforeEach(() => {
+  jest.clearAllMocks();
+  (globalThis as any).fetch = mockFetch;
```

Comment:
The tests overwrite globalThis.fetch but never restore the original fetch, which can leak into other tests. Save the original fetch and restore it in afterAll/afterEach.
```
const realFetch = globalThis.fetch;
beforeEach(() => { globalThis.fetch = mockFetch; });
afterAll(() => { globalThis.fetch = realFetch; });
```